server/setting.go

```diff
@@ -60,6 +61,31 @@ func (s *Server) registerSettingRoutes(g *echo.Group) {
 			return echo.NewHTTPError(http.StatusBadRequest, "Malformed update setting request").SetInternal(err)
 		}
 
+		if settingPatch.Name == api.SettingAppIM {
+			var value api.SettingAppIMValue
+			if err := json.Unmarshal([]byte(settingPatch.Value), &value); err != nil {
+				return err
+			}
+			if value.IMType != (api.IMTypeFeishu) {
+				return errors.New("unknown IM type")
+			}
+			p := s.ApplicationRunner.P
+			approvalCode, err := p.CreateApprovalDefinition(ctx, feishu.TokenCtx{
+				AppID:     value.AppID,
+				AppSecret: value.AppSecret,
+				Token:     p.Token.Load().(string),
```

Comment:
The token here should be implementation detail that shouldn't be accessed outside of the package "feishu".